### PR TITLE
[IOS-693]: Add validators for other syntaxes

### PR DESCRIPTION
**Pull request by kkapitan**
### Ticket

https://netguru.atlassian.net/browse/IOS-693

### Description

This PR introduces validation for new string syntaxes:
- Separation of `http` and `https`
- Websocket (`ws` and `wss`)
- IPv4 and IPv6
- Domains (like `example.com`)
- UUID
- Geo coordinates
- Price
- ISBN
- Color in hex
- Phone number
- Postal code

### CODE REVIEW

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 1]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isGeoCoordinate {
+    NSString *pattern = @"-?\\d{1,3}\\.\\d+";
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isPrice {
+    NSArray<NSString *> *patterns = @[
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: How about using some sort of `static`?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 2]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isUUID {
+    NSString *pattern = @"^[0-9A-Fa-f]{8}\\-[0-9A-Fa-f]{4}\\-[0-9A-Fa-f]{4}\\-[0-9A-Fa-f]{4}\\-[0-9A-Fa-f]{12}$";
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isGeoCoordinate {
+    NSString *pattern = @"-?\\d{1,3}\\.\\d+";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe `static`?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 3]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isDomain {
+    NSString *pattern = @"^([a-zA-Z0-9]([a-zA-Z0-9\\-]{0,61}[a-zA-Z0-9])?\\.)+[a-zA-Z]{2,6}$";
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isUUID {
+    NSString *pattern = @"^[0-9A-Fa-f]{8}\\-[0-9A-Fa-f]{4}\\-[0-9A-Fa-f]{4}\\-[0-9A-Fa-f]{4}\\-[0-9A-Fa-f]{12}$";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 4]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isIPv6 {
+    NSString *pattern = @"((^|:)([0-9a-fA-F]{0,4})){1,8}$";
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isDomain {
+    NSString *pattern = @"^([a-zA-Z0-9]([a-zA-Z0-9\\-]{0,61}[a-zA-Z0-9])?\\.)+[a-zA-Z]{2,6}$";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 5]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    return [self ngr_isURLWithScheme:@"wss"];
+}
+
+- (BOOL)ngr_isIPv4 {
+    NSString *pattern = @"^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$";
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isIPv6 {
+    NSString *pattern = @"((^|:)([0-9a-fA-F]{0,4})){1,8}$";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 6]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+- (BOOL)ngr_isWebSocketURL {
+    return [self ngr_isURLWithScheme:@"ws"];
+}
+    
+- (BOOL)ngr_isSecureWebSocketURL {
+    return [self ngr_isURLWithScheme:@"wss"];
+}
+
+- (BOOL)ngr_isIPv4 {
+    NSString *pattern = @"^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 7]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    NSArray<NSString *> *patterns = @[
+        @"\\d+(\\.\\d{2})?",
+        @"\\d+(,\\d{2})?",
+    ];
+    
+    return [self ngr_evaluateAnyOfPatterns:patterns];
+}
+    
+- (BOOL)ngr_isISBN {
+    NSString *pattern = @"(?:(?=.{17}$)97[89][ -](?:[0-9]+[ -]){2}[0-9]+[ -][0-9]|97[89][0-9]{10}|(?=.{13}$)(?:[0-9]+[ -]){2}[0-9]+[ -][0-9Xx]|[0-9]{9}[0-9Xx])";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 8]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isHexColor {
+    NSString *pattern = @"^#?([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 9]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isPhoneNumber {
+    NSString *pattern = @"^((\\+)|(00))[0-9]{6,14}$";
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 10]
```diff
@@ -44,18 +44,97 @@ - (BOOL)ngr_isURLWithScheme:(NSString *)scheme {
...
+
+    return [self ngr_evaluatePattern:pattern];
+}
+    
+- (BOOL)ngr_isPostalCode {
+    NSArray<NSString *> *patterns = @[
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Maybe static?

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m [hunk 11]
```diff
@@ -111,11 +190,21 @@ - (BOOL)ngr_isValidCreditCardExpirationDate {
 }
 
 #pragma mark - Private
-
+
```
Comment (rad3ks) [inline NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: 👀

File: NGRValidatorTests/Helpers/NGRSyntaxTestData.h [hunk 12]
```diff
@@ -0,0 +1,22 @@
+//
+//  NGRSyntaxTestData.h
+//  NGRValidator
+//
+//  Created by Krzysztof Kapitan on 27.03.2017.
```
Comment (rad3ks) [inline NGRValidatorTests/Helpers/NGRSyntaxTestData.h]: This is not needed

**Review by rad3ks — COMMENTED** [review]
Only few minor issues

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m (hunk 6 above)
Comment (kkapitan) [reply NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: Good idea! Thanks

File: NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m (hunk 11 above)
Comment (kkapitan) [reply NGRValidator/NGRValidator/Categories/NSString+NGRValidator.m]: 🙈

**Review by rad3ks — APPROVED** [review]

**Pull request merged**